### PR TITLE
 add content disposition for OK responses 

### PR DESCRIPTION
**Pull request by rentenmark**
following on to discussions from #2541 this is a smaller PR to fix up the OK responses only

### CODE REVIEW

**Comment by ignopeverell** [comment]
While we're cleaning this, shouldn't those return `204`s instead?

**Comment by kargakis** [comment]
> While we're cleaning this, shouldn't those return `204`s instead?

You may break more people than it's worth by changing this now.

File: wallet/src/controller.rs [hunk 1]
```diff
@@ -677,20 +677,27 @@ fn create_ok_response(json: &str) -> Response<Body> {
...
-		)
-		.body(text.into())
-		.unwrap()
+		);
+
+	if status == StatusCode::OK {
+		builder = builder.header(hyper::header::CONTENT_TYPE, "application/json");
+	}
+
+	builder.body(text.into()).unwrap()
```
Comment (garyyu) [inline wallet/src/controller.rs]: Perhaps add one line of comment here to remind the caller of this `response` function, the `text` here is required to be a JSON format.

Or even we can change the variable name `text` here as for example `json_formated_text`?

**Comment by ignopeverell** [comment]
@kargakis `<html mode:pedantic="false" xmlns:mode="http://www.w3.org/1999/xhtml/pedantism">Alright then</html> `

**Pull request merged**